### ISSUE TITLE
lazy load dependencies

### ISSUE THREAD

**Issue by erjoalgo** [skarra/ASynK#116]
I'm not interested in syncing to/from google, so I shouldn't need the google-contacts-api dependency. For the same reason I don't really need an oauth client, for example.

Can we make dependencies be loaded lazily depending on what type of conversion is requested?

**Comment by skarra** [comment]
You are right; it should be possible. In fact there is a rarely used feature of ASynK to sync with MS Outlook on Windows (the first connector I made actually) which works on a 'lazy load' style.

This is a low priority fix for me at this time.
